canonical-web-and-design/maas-ui
# feat(ui): surface allocated_other data in KVM and cluster details pages

## Done

- Replaced `PodMeter` component with new (and improved!) `KVMResourceMeter` component which displays the split between allocated, other and free amounts of the resource. This component is reused across the KVM list and KVM details pages.
- Updated the prop shapes of a few of the `Resources` components so it's a bit more generic.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the LXD KVM list and take note of the usage values for cores, RAM and storage of a LXD single host.
- Go to the "Virtual machines" tab of that host and check that the values displayed here are the same as those displayed in the list.
- Go to the "Resources" tab and check that the values are again the same, and the values for each storage pool is correct.
- Check the same things for a LXD cluster as well as a virsh host.
- Open `ui/src/app/kvm/components/StorageResources/StorageResources.tsx` and edit line 24 so it says `!==` instead of `===`
- Open `ui/src/app/kvm/components/LXDVMsSummaryCard/LXDVMsSummaryCard.tsx` and edit lines 50-52 to:
  ```
  hugepagesAllocated={1024}
  hugepagesFree={2048}
  hugepagesOther={4096}
  ```
- Go back to the "Virtual machines" tab of the LXD KVM and check that a row for hugepages shows underneath the RAM chart.
- Hover over the segments and check that the tooltip includes the hugepage amount
- Check that the Storage section shows as cards and the values are correct.

## Fixes

Fixes canonical-web-and-design/app-squad#484

## Screenshots
### VMs tab
![Screenshot 2021-11-05 at 14-22-24 bolla virtual machines bolla MAAS](https://user-images.githubusercontent.com/25733845/140458131-b938c553-1e6f-4ea5-8405-75699f7d3470.png)


### Resources tab
![Screenshot 2021-11-05 at 14-22-36 LXD resources bolla bolla MAAS](https://user-images.githubusercontent.com/25733845/140458141-8d41db1c-0883-425b-8637-952690e272d0.png)
